keel-core/src/main/kotlin/com/netflix/spinnaker/keel/events/NotificationEventListener.kt

```diff
@@ -0,0 +1,75 @@
+package com.netflix.spinnaker.keel.events
+
+import com.netflix.spinnaker.keel.api.NotificationType
+import com.netflix.spinnaker.keel.core.api.PromotionStatus
+import com.netflix.spinnaker.keel.persistence.KeelRepository
+import org.slf4j.LoggerFactory
+import org.springframework.context.ApplicationEventPublisher
+import org.springframework.context.event.EventListener
+import org.springframework.stereotype.Component
+import java.time.Clock
+
+/**
+ * Responsible to listen to notification events, and fetch the information needed
+ * for sending a notification, based on NotificationType.
+ */
+@Component
+class NotificationEventListener(
```

Comment:
encapsulate all notifications related information into this class.

Reply:
I think this is definitely an improvement.

The question I have is: why not move this logic directly to the `keel.slack.notifications.*` event listeners functions, so that those functions listen directly for `PinnedNotification` events?

Reply:
I thought it might be better to separate the logic between two pieces - one (NotificationEventListener) will be responsible to fetch the relevant data, and the second (events under the `slack` module) will just construct the slack specific notification per event. 
I'm thinking of a couple of things that might support this theory - 
1. we can utilize it to send emails in the future
2. we need a central place to control which notifications are being sent according to user-defined frequency, and I _think_ having it under `keel-core` will make more sense than under `keel-slack` module.

I can also refactor it in the near future if we feel it's overhead, as I still need to do a bunch of changes in the code. Let me know if it makes sense to you, we can also have a quick chat about it.

Reply:
I thought about it a lot and I agree. It doesn't make sense to store in two places the data needed for each event. Will move it directly to the `keel-slack` module.